tests/integration/bond_test.py

```diff
@@ -403,3 +406,77 @@ def test_change_bond_option_miimon(bond99_with_2_slaves):
     bond_options["miimon"] = "200"
     libnmstate.apply(desired_state)
     assertlib.assert_state(desired_state)
+
+
+def test_create_bond_without_mode():
+    with bond_interface(name=BOND99, slaves=[], create=False) as state:
+        state[Interface.KEY][0][Bond.CONFIG_SUBTREE].pop(Bond.MODE)
+        with pytest.raises(NmstateValueError):
+            libnmstate.apply(state)
+
+
+def test_bond_mac_restriction_without_mac_in_desire(eth1_up, eth2_up):
+    with bond_interface(
+        name=BOND99,
+        slaves=[ETH1, ETH2],
+        extra_iface_state={
+            Bond.CONFIG_SUBTREE: {
+                Bond.MODE: BondMode.ACTIVE_BACKUP,
+                Bond.OPTIONS_SUBTREE: {"fail_over_mac": "active"},
+            },
+        },
+    ) as state:
+        assertlib.assert_state_match(state)
+
+
+def test_bond_mac_restriction_with_mac_in_desire(eth1_up, eth2_up):
+    with bond_interface(
+        name=BOND99,
+        slaves=[ETH1, ETH2],
+        extra_iface_state={
+            Interface.MAC: MAC0,
+            Bond.CONFIG_SUBTREE: {
+                Bond.MODE: BondMode.ACTIVE_BACKUP,
+                Bond.OPTIONS_SUBTREE: {"fail_over_mac": "active"},
+            },
+        },
+        create=False,
+    ) as state:
+        with pytest.raises(NmstateValueError):
+            libnmstate.apply(state)
+
+
+def test_bond_mac_restriction_in_desire_mac_in_current(bond99_with_2_slaves):
```

Comment:
I don't understand the test name. What is tested here?

Reply:
It tests user want to switch to the mac restriction mode, and don't want to be failed as MAC address exists in current status.

Reply:
I realized later that it would make more sense to me to call this "test_active_fail_over_mac_with_mac_in_current" or similar to specify the option that is tested.